src/config.rs

```diff
@@ -27,32 +27,38 @@ pub struct ServerConfig {
 #[derive(Deserialize, Debug)]
 pub struct RpcMethod {
     pub method: String,
-    pub params: Vec<MethodParam>,
+
+    #[serde(default)]
+    pub params: Vec<ParamType>,
 
     #[serde(default)]
     pub cache: usize,
 }
 
 impl RpcMethod {
-    pub fn inject_block_num(&self) -> Option<usize> {
+    pub fn at(&self) -> Option<(usize, &ParamType)> {
         self.params
             .iter()
-            .position(|p| p.inject == Some(true) && p.ty == "BlockNumber")
-    }
-
-    pub fn inject_block_hash(&self) -> Option<usize> {
-        self.params
-            .iter()
-            .position(|p| p.inject == Some(true) && p.ty == "BlockHash")
+            .enumerate()
+            .rfind(|(_index, param)| matches!(param, ParamType::At(_)))
     }
 }
 
-#[derive(Deserialize, Debug, Eq, PartialEq)]
-pub struct MethodParam {
-    pub name: String,
-    pub ty: String,
-    pub is_optional: Option<bool>,
-    pub inject: Option<bool>,
+#[derive(Deserialize, Debug, Clone)]
+pub enum ParamType {
+    U32,
```

Comment:
this is interesting. but I don't want to limit the ParamType. user should be able to specify whatever type and we just ignore ones we don't understand.
maybe somehow make it default to `Type(Strinig)`?

Reply:
We can have Any which ignores validation

Reply:
@xlc Any type should be enough. We don't need exact type if you're not validating anything

Reply:
Let's only keep types that we are actually using

Reply:
ok removing Any

Reply:
I mean only keep Any, BlockHash, and BlockNumber

Reply:
Ok